blob/service_test.go

```diff
@@ -0,0 +1,234 @@
+package blob
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"sort"
+	"testing"
+	"time"
+
+	ds "github.com/ipfs/go-datastore"
+	ds_sync "github.com/ipfs/go-datastore/sync"
+	mdutils "github.com/ipfs/go-merkledag/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	"github.com/celestiaorg/go-header/store"
+	"github.com/celestiaorg/nmt/namespace"
+
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/header/headertest"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/getters"
+)
+
+func TestService_GetSingleBlob(t *testing.T) {
```

Comment:
A lot of these tests could probably be combined and run with t.Run so you only need one set up, but more of a stylistic thing. Can keep as is if wanted

Reply:
Yeah, this makes sense. Will rework tests